tests/PipelineTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Bugsnag\Tests;
+
+use Bugsnag\Pipeline;
+use PHPUnit_Framework_TestCase as TestCase;
+
+class TestCallbackA
+{
+    public function __invoke($item, $next)
+    {
+        $payload = [
+            'B' => array_key_exists('B', $item),
+            'C' => array_key_exists('C', $item),
+        ];
+        $item['A'] = $payload;
+        $next($item);
+    }
+}
+
+class TestCallbackB
+{
+    public function __invoke($item, $next)
+    {
+        $payload = [
+            'A' => array_key_exists('A', $item),
+            'C' => array_key_exists('C', $item),
+        ];
+        $item['B'] = $payload;
+        $next($item);
+    }
+}
+
+class TestCallbackC
+{
+    public function __invoke($item, $next)
+    {
+        $payload = [
+            'A' => array_key_exists('A', $item),
+            'B' => array_key_exists('B', $item),
+        ];
+        $item['C'] = $payload;
+        $next($item);
+    }
+}
+
+class PipelineTest extends TestCase
+{
+    public function testCallableAddedToPipeline()
+    {
+        $pipeline = new Pipeline();
+        $pipeline->pipe(new TestCallbackA());
+        $pipeline->execute([], function ($item) {
```

Comment:
These tests seem overly complex to me. Why not make the tests append to a string? so the string at the end would spell out 'abc' or 'bac' if b was run first?